flowgate-api/src/main/java/com/vmware/flowgate/controller/AssetController.java

```diff
@@ -548,9 +549,57 @@ public ServerMapping getMappingById(@PathVariable("id") String id) {
    @RequestMapping(value = "/mapping/hostnameip", method = RequestMethod.POST)
    public void createHostNameIPMapping(@RequestBody AssetIPMapping mapping) {
       BaseDocumentUtil.generateID(mapping);
+      String ip = mapping.getIp();
+      if(ip == null || ip.isEmpty() || !ip.matches(HandleURL.IP_REGX)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid ip address",
+               null);
+      }
+      if (!assetService.isAssetNameValidate(mapping)) {
+         throw new WormholeRequestException(HttpStatus.INTERNAL_SERVER_ERROR, "Invalid asset name",
```

Comment:
let' change the error message more meanful.

Reply:
Done